app/src/main/java/org/oppia/app/topic/questionplayer/QuestionPlayerFragmentPresenter.kt

```diff
@@ -163,7 +163,7 @@ class QuestionPlayerFragmentPresenter @Inject constructor(
 
   fun onResponsesHeaderClicked() {
     recyclerViewAssembler.togglePreviousAnswers(questionViewModel.itemList)
-    recyclerViewAssembler.adapter.notifyDataSetChanged()
+    recyclerViewAssembler.adapter.setDataUnchecked(questionViewModel.itemList)
```

Comment:
Per https://github.com/oppia/oppia-android/blob/0577e57ee6f0541561f70becfa76ea4f39b5e048/app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt#L174 it seems like the state fragment doesn't need to do this. Why? What's different about questions? I'm wondering if there's something else fundamentally broken here that needs to be fixed. This change might just be avoiding the issue by leveraging DiffUtil (which might be better in the long-term for this recyclerview, anyway), and we should understand that the underlying issue is in detail. If you know, please add context here (and as a comment since we don't want people changing this in the future).

Please also add tests to verify the behavior has been fixed & stays fixed. These tests should pass with your change and fail without it.

Reply:
It was because of the wrong import of `ObservableArrayList`.
We can add test cases regarding the same can go under #503

Reply:
We actually do have the ability to test the question player in isolation. See StateFragmentLocalTest. Could you add a test that would break without this fix?

Also to be clear, the issue isn’t the import: it’s why the wrong import caused the failure and how. The changed import is the fix, not the problem.

Reply:
Ok, I will add a test case regarding `org.oppia.app.viewmodel.ObservableArrayList removeAll()` method and check on how does this test case can fail if not used properly.

Reply:
Suggest adding three tests:
- Submit wrong answer & verify answers are collapsed
- Submit wrong answer, expand answers, verify the wrong answer is expanded
- Submit wrong answer, expand & collapse answer, and verify that answer is now collapsed again

Reply:
Done